### PR TITLE
Add coverage

### PR DESCRIPTION
**Pull request by typhoonzero**
Add go code coverage tools

NOTE that coverage is only updated after nighly build

Another change added in this PR is to fix CI random fails at: https://github.com/sql-machine-learning/sqlflow/pull/369/files#diff-c6c4a707bc1aeb863798a5f102630ff0R60

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,21 +1,14 @@
 language: go
-
 branches:
```
Comment (tonyyang-svail) [inline .travis.yml:3]: Indentation changes?

File: cmd/sqlflowserver/main_test.go [hunk 2]
```diff
@@ -57,6 +57,15 @@ func AssertEqualAny(a *assert.Assertions, expected interface{}, actual *any.Any)
 	}
 }
 
+func AssertGreaterEqualAny(a *assert.Assertions, actual *any.Any, expected interface{}) {
```
Comment (tonyyang-svail) [inline cmd/sqlflowserver/main_test.go:60]: I don't think the following changes are relevant to this PR.

File: .travis.yml [hunk 3]
```diff
@@ -33,10 +26,12 @@ script:
...
-
 deploy:
   skip_cleanup: true
   provider: script
   script: bash scripts/deploy.sh
   on:
     branch: develop
+env:
+  global:
+    secure: f03CHRo2Afl9PD6jLuKtY/6b/MgJkytcfeIhSrAYZFLzUCQTa4MI5yih4pG/VJtTpKVpoQD7NJikj/x0eO/6NOGxegJwaruY8EDg/4LVLeuVqRipmw0V9l3LcZ7AofxPAeJ6RZLpHALR9cYegXEEtvyfjNRw2BhQZsZBvTE8sbbvfzLJOY1UC9CeQnHxDGZYXG3WSMr7b+44ifg4OBweATmEsdIVPD7kgexunJvHw0vzQ25vsf+huYwLorVc7uRghoYY8SoaLMEcmP50Jsl2Mn7AUoFUyM3tl/kOub11BV4/yGrGJC1BysT7l48frSKYqSU1yeT+0xiJjgkOwpV8BcxOmXrgm2s6PDzWDmiXtBRhE6n79iG2IuuTuYLk4n/Rm7uwKYu6EdnooKSnhGV6DW7mLVJ5Rcb655xCB+AoXb79Ilp60YxsMFTqzdhMkf3QpQV/OsEHFrzNhab6+pWcrwOeCq/CXU0MDO0rQuX4YIBwdEyAlDrj2JDvJ868lpdTJTxUUgUovH2j8rFrtCf5Bx0gqg0Cj81s/7cMdhtoV+wbTtHoQ7TLTZkvqYyFAMCkl2d5g00UhZpq+HTjzcPOL5xKc0mZhsuYq03nTVMQWzIJiKkaWbDPoY3eZcLga0cO8VD2HgzsQ5lg300ZTF73AxonZheyzJVWtLQvrpT8dPI=
```
Comment (tonyyang-svail) [inline .travis.yml:43]: What is the purpose of `env.global.secure`?

File: Dockerfile.dev [hunk 4]
```diff
@@ -39,9 +39,11 @@ RUN rm protoc-3.6.1-linux-x86_64.zip
 RUN mkdir -p /go
 ENV GOPATH /go
 ENV PATH $PATH:/usr/local/go/bin:/go/bin
-
 RUN go get github.com/golang/protobuf/protoc-gen-go && \
-go get -u golang.org/x/lint/golint
+go get golang.org/x/lint/golint && \
+go get golang.org/x/tools/cmd/cover && \
+go get github.com/mattn/goveralls
```
Comment (tonyyang-svail) [inline Dockerfile.dev]: As commented in `deploy.sh`, shall we delete all `go get`s in the Dockerfile?

**Review by tonyyang-svail — CHANGES_REQUESTED** [review]

**Comment by tonyyang-svail** [comment]
> NOTE that coverage is only updated after the nightly build

@typhoonzero  Where can we find the coverage percental?

File: .travis.yml (hunk 3 above)
Comment (typhoonzero) [reply .travis.yml:43]: Please check out https://docs.coveralls.io/go for details about this.

**Review by wangkuiyi — DISMISSED** [review]

File: cmd/sqlflowserver/main_test.go (hunk 2 above)
Comment (typhoonzero) [reply cmd/sqlflowserver/main_test.go:60]: I updated the description of this PR, this change try to fix randomly fails in PR CI check, or pull requests would take several re-runs to pass the checks.

File: Dockerfile.dev (hunk 4 above)
Comment (typhoonzero) [reply Dockerfile.dev]: Sure, Done.

**Comment by typhoonzero** [comment]
> @typhoonzero Where can we find the coverage percental?

In https://coveralls.io/

File: .travis.yml (hunk 1 above)
Comment (typhoonzero) [reply .travis.yml:3]: Done.

**Review by tonyyang-svail — APPROVED** [review]
LGTM!

**Pull request merged**